src/cljs/athens/db.cljs

```diff
@@ -280,16 +280,18 @@
 
 
 (defn search-in-node-title
-  [query]
-  (d/q '[:find [(pull ?node [:db/id :node/title :block/uid]) ...]
-         :in $ ?query-pattern ?query
-         :where
-         [?node :node/title ?title]
-         [(re-find ?query-pattern ?title)]
-         [(not= ?title ?query)]] ;; ignore exact match to avoid duplicate
-       @dsdb
-       (re-case-insensitive query)
-       query))
+  ([query] (search-in-node-title query 20))
```

Comment:
Do we need another overload to find all results (e.g. for custom components/for toggling between top-20 and all?)

Reply:
> I think this refactor makes sense in terms of performance - but I think another question concerning this UX-wise is whether we show only the top 20 results or show everything in the search dropdown from `keybinding.cljs`. What do you think @shanberg @tangjeff0 ?

@HaojiXu I think it makes sense to enforce a default cap on search results for any dropdown because depending on the search query, there could be hundreds of results and we probably don't want to display all of them to the user. And if we require more than 20 results, we can specify a higher max number of results.

Reply:
Definitely. I do believe that it might be a bit tricky to determine which results should be in the top 20, but just as you said, this is a question we can talk about later.